README.md

```diff
@@ -1,26 +1,55 @@
-# Consensus Compliance tests
+# Consensus Compliance Tests
+This repository generates test cases for the Ethereum [Proof-of-Stake Consensus Spec](https://github.com/ethereum/consensus-specs) for fork choice compliance testing.
 
-## instructions
-```
-# using my repo currently, fc-compliance2 branch
-git clone https://github.com/ericsson49/eth2.0-specs.git
-cd eth2.0-specs
-git checkout fc-compliance2
+## Dependencies
+- Python 3.12+
+- System dependencies:
+  - gcc
+  - libsnappy-dev
+  - libssl-dev
+  - python3-dev
+  - build-essential
 
-# setup python stuff
 
-cd tests/generators/fork_choice_generated
-python3 -m venv venv
-. venv/bin/activate
-# don't need it for tests themselves, but it's required to install requirements
-pip install wheel
+## Test generation
 
-# turn on multiprocessing, do not have a command line switch at the moment
-echo '\nGENERATOR_MODE = MODE_MULTIPROCESSING\n' >> ../../core/pyspec/eth2spec/gen_helpers/gen_base/settings.py
+### Manual
+```bash
+# Clone the eth2.0-specs repository
+git clone -b fc-compliance2 https://github.com/ericsson49/eth2.0-specs.git
+cd eth2.0-specs/tests/generators/fork_choice_generated
 
+# Setup Python environment
+python3 -m venv .venv
+source .venv/bin/activate
 pip install -r requirements.txt
 
+# turn on multiprocessing, do not have a command line switch at the moment
+echo '\nGENERATOR_MODE = MODE_MULTIPROCESSING\n' >> ../../core/pyspec/eth2spec/gen_helpers/gen_base/settings.py
+
+# Generate test cases
 mkdir tests_tiny
 python test_gen.py -o tests_tiny --fc-gen-config tiny/test_gen.yaml
 ```
 
+### CI/CD
+Test cases are generated on:
+- Every push to `master`
+- Pull requests
+- Manual trigger via GitHub Actions UI
+
+### Manual Workflow Trigger Options
+
+You can manually trigger the workflow with custom parameters:
+
+- **specs_repo**: eth2.0-specs repository URL (default: `ericsson49/eth2.0-specs`)
+- **specs_ref**: Branch, tag or commit to use (default: `fc-compliance2`)
+- **specs_path**: Local path to clone the specs (default: `eth2.0-specs`)
+
+## Usage
+Generated test cases are available as downloadable artifacts from the GitHub Actions workflow runs. 
+
+1. Move into the scripts directory: `cd scripts`
+2. Download and unzip the test cases from the workflow run
+3. Install the requirements: `pip3 install ruamel.yaml==0.17.21 wheel setuptools && pip3 install -r requirements.txt`
```

Comment:
It shouldn't be necessary to install `ruamel.yaml==0.17.21` here as it's listed in the requirements.txt file.
```suggestion
3. Install the requirements: `pip3 install wheel setuptools && pip3 install -r requirements.txt`
```

Reply:
weirdly enough, both in the CI and on my builder machine I needed to explicitly install it again :/ I'm honestly not sure why :/